lib/bloomFilterIO/bloomFilterReader.ts

```diff
@@ -1,22 +1,17 @@
 import parquet_util from "../util";
 import parquet_thrift from "../../gen-nodejs/parquet_types";
 import sbbf from "../bloom/sbbf";
-import { ParquetEnvelopeReader } from "parquet";
-
+import { ParquetEnvelopeReader } from "../reader"
 import { ColumnChunkData } from "../types/types";
 
 const filterColumnChunksWithBloomFilters = (
   columnChunkDataCollection: Array<ColumnChunkData>
 ) => {
   return columnChunkDataCollection.filter((columnChunk) => {
-    const {
-      column: {
-        meta_data: {
-          bloom_filter_offset: { buffer: bloomFilterOffsetBuffer },
-        },
-      },
-    } = columnChunk;
-    return bloomFilterOffsetBuffer;
+    if (!columnChunk.column.meta_data.bloom_filter_offset) {
```

Comment:
the previous form was causing an error in the browser when meta_data was undefined.

Reply:
This should cause an error everywhere if `columnChunk.column.meta_data` is undefined. Did you mean to say when `bloom_filter_offset` is undefined? Also this should be redundant to the optional chain below unless you're really checking for falsey rather than undefined here. I would drop it.

Reply:
Now that I think it was actually bloom_filter_offset, and somehow  the case I was looking at was getting through the if condition.